process/block/shardblock.go

```diff
@@ -250,6 +254,44 @@ func (sp *shardProcessor) ProcessBlock(
 	return nil
 }
 
+func (sp *shardProcessor) setShardConsensusData(headerHandler data.HeaderHandler) error {
```

Comment:
this functions should be in a separate component.

Reply:
the logic for selecting the right addresses etc was moved to the address handler, these simply call the interface, so there is no need of creating a separate component.